modules/gdpr_tasks/src/Traversal/RightToAccessEntityTraversal.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Drupal\gdpr_tasks\Traversal;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\FieldableEntityInterface;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\gdpr_fields\Entity\GdprFieldConfigEntity;
+use Drupal\gdpr_fields\EntityTraversal;
+
+/**
+ * Entity traversal for performing Right to Access requests.
+ *
+ * @package Drupal\gdpr_tasks
+ */
+class RightToAccessEntityTraversal extends EntityTraversal {
+  private $assets;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function traverse(EntityInterface $entity) {
+    $this->assets = [];
+    $results = parent::traverse($entity);
+    $results['_assets'] = $this->assets;
+    return $results;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function processEntity(FieldableEntityInterface $entity, GdprFieldConfigEntity $config, $row_id, array &$results) {
+    $entity_type = $entity->getEntityTypeId();
+
+    $fields = $this->entityFieldManager->getFieldDefinitions($entity_type, $entity->bundle());
+    $field_configs = $config->getFieldsForBundle($entity->bundle());
+
+    foreach ($fields as $field_id => $field) {
+      $field_config = isset($field_configs[$field_id]) ? $field_configs[$field_id] : NULL;
+
+      // If the field is not configured, not enabled,
+      // or not enabled for RTA, then skip it.
+      if ($field_config === NULL || !$field_config->enabled || !in_array($field_config->rta, ['inc', 'maybe'])) {
+        continue;
+      }
+
+      $plugin_name = "{$entity_type}|{$entity->bundle()}|{$field_id}";
+      $filename = empty($field_config->sarsFilename) ? 'main' : $field_config->sarsFilename;
```

Comment:
Unfortunately it's not quite as simple as this, as the filename is based on the source, which could be an entity reference on another entity or an owner on this entity... In D7 I was also passing around the source plugin :'(

Reply:
Perhaps one solution to this would be to use a [non shared service](https://symfony.com/doc/current/service_container/shared.html) (so each call creates a new instance). Then we can use properties on the object much more safely. That may make the whole arguments stuff much simple and could have a source stack or similar to track where we are and how things relate.

Reply:
I think the new `$parent_config` parameter should handle this? //cc @yanniboi

Reply:
Done